src/profile-logic/marker-schema.js

```diff
@@ -0,0 +1,217 @@
+      { key: 'stylesShared', label: 'Styles shared', format: 'integer' },
+      { key: 'stylesReused', label: 'Styles reused', format: 'integer' },
+    ],
+  },
+  {
+    name: 'PreferenceRead',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'prefName', label: 'Name', format: 'string' },
+      { key: 'prefKind', label: 'Kind', format: 'string' },
+      { key: 'prefType', label: 'Type', format: 'string' },
+      { key: 'prefValue', label: 'Value', format: 'string' },
+    ],
+  },
+  {
+    name: 'Invalidation',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'url', label: 'URL', format: 'url' },
+      { key: 'line', label: 'Line', format: 'integer' },
+    ],
+  },
+  {
+    name: 'UserTiming',
+    display: ['marker-chart', 'marker-table'],
+    data: [{ key: 'name', label: 'Name', format: 'string' }],
+  },
+  {
+    name: 'Text',
+    display: ['marker-chart', 'marker-table'],
+    data: [{ key: 'name', label: 'Name', format: 'string' }],
+  },
+  {
+    name: 'Log',
+    display: ['marker-table'],
+    data: [
+      { key: 'module', label: 'Module', format: 'string' },
+      { key: 'name', label: 'Name', format: 'string' },
+    ],
+  },
+  {
+    name: 'Bailout',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'bailoutType', label: 'Type', format: 'string' },
+      { key: 'where', label: 'Where', format: 'string' },
+      { key: 'script', label: 'Script', format: 'string' },
+      { key: 'functionLine', label: 'Function Line', format: 'string' },
+      { key: 'bailoutLine', label: 'Bailout Line', format: 'string' },
+    ],
+  },
+  {
+    name: 'tracing',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      // This is really the "type" of the marker.
+      { key: 'category', label: 'Category', format: 'string' },
+    ],
+  },
+  {
+    name: 'IPC',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'messageType', label: 'Type', format: 'string' },
+      { key: 'sync', label: 'Sync', format: 'string' },
+    ],
+  },
+];
+
+export function formatFromMarkerSchema(
+  markerType: string,
+  format: MarkerFormatType,
+  value: any
+): string | null {
+  switch (format) {
+    case 'url':
+    case 'file-path':
+    case 'string':
+      return value;
+    case 'duration':
+    case 'time':
+      return formatTimestamp(value);
+    case 'seconds':
+      return formatSeconds(value);
+    case 'milliseconds':
+      return formatMilliseconds(value);
+    case 'microseconds':
+      return formatMicroseconds(value);
+    case 'nanoseconds':
+      return formatNanoseconds(value);
+    case 'bytes':
+      return formatBytes(value);
+    case 'integer':
+      return formatNumber(value, 0, 0);
+    case 'decimal':
+      return formatNumber(value);
+    case 'percentage':
+      return formatPercent(value);
+    case 'stack':
+      return null;
```

Comment:
Would be good to add a comment that says we have a special handling for this.

Reply:
I actually removed it, since I'm not supporting it yet.